lib/pg_ha_migrations/struct.rb

```diff
@@ -0,0 +1,134 @@
+module PgHaMigrations::Struct
+  Relation = Struct.new(:name, :schema) do
+    def self.connection
+      ActiveRecord::Base.connection
+    end
+
+    delegate :inspect, to: :name
+    delegate :connection, to: :class
+
+    def fully_qualified_name
+      @fully_qualified_name ||= [
+        PG::Connection.quote_ident(schema),
+        PG::Connection.quote_ident(name),
+      ].join(".")
+    end
+
+    def present?
+      name.present? && schema.present?
+    end
+  end
+
+  class Table < Relation
+    def self.from_table_name(table)
+      pg_name = ActiveRecord::ConnectionAdapters::PostgreSQL::Utils.extract_schema_qualified_name(table.to_s)
+
+      schema_conditional = if pg_name.schema
+        "#{connection.quote(pg_name.schema)}"
+      else
+        "ANY (current_schemas(false))"
+      end
+
+      schema = connection.select_value(<<~SQL)
+        SELECT schemaname
+        FROM pg_tables
+        WHERE tablename = #{connection.quote(pg_name.identifier)} AND schemaname = #{schema_conditional}
+        ORDER BY array_position(current_schemas(false), schemaname)
+        LIMIT 1
+      SQL
+
+      raise ArgumentError, "Could not find table #{table}" unless schema.present?
```

Comment:
This was previously a migration error, but it seems weird to raise that since this class can be used outside of migrations. Maybe we need to add a new error class?

Reply:
That's a good point. Probably still worth a PgHaMigrations scoped error class IMO.

Reply:
Addressed in https://github.com/braintree/pg_ha_migrations/pull/87/commits/bf4407d629c2b4eb8e5f1d23f85f9a8087e8fd06